docs/development/design/automation-rules.rst

```diff
@@ -0,0 +1,90 @@
+Automation Rules
+================
+
+Automation rules allow users to apply some actions over new or deleted versions.
+See :doc:`/automation-rules` for more details of the current features.
+
+This document explore some ideas to expand automation rules to automate more tasks.
+
+Current state and limitations
+-----------------------------
+
+- Matches are done using regular expressions.
+  But the models were designed so other type of matches can be implemented
+  (like comparing the version with the highest semver version).
+- Automation rules are meant to automate existing actions,
+  so an action shouldn't be able to be triggered only inside an automation rule.
+  This is so users aren't blocked if a rule didn't match a version,
+  or if they need to execute the action over existing versions.
+- Automation rules are currently matched against new versions,
```

Comment:
I wonder if a better way to improve this might be removing the above restriction (eg. UI must be able to do it) and allow users to run automation rules across all their existing versions. This seems like a useful improvement.

Reply:
A problem with that is if the user wants to change only one version or a couple, it needs to create a new rule with a pattern that only matches those versions, then run that rule against existing versions and delete it so new versions use a new rule.

We also need to think about how to place this element in the UI, but I guess that's the same amount of work (or maybe less) if we want to add a feature to enable it manually.

Reply:
"Apply an automation rule on existing versions", is that what we are taking about here? If so, isn't it very similar to "Live feedback of versions that match a rule" feature? I mean, it seems it's exactly the same code. We could have a checkbox saying "Apply this rule to existing versions" when creating the automation rule.

Reply:
yeah, but that has the problem that you'll need to write several rules and delete them just to match a couple of versions, instead of having the option of editing one by one, or in order to edit something you need to create a new rule. But I can see where that option could be useful too if you want to edit versions in bulk.